keg_storage/tests/test_backend_azure.py

```diff
@@ -4,29 +4,94 @@
 import string
 import urllib.parse as urlparse
 from io import BytesIO
+from typing import Union
 from unittest import mock
+import urllib.parse
 
 import arrow
 import pytest
-from azure.storage.blob import BlobProperties
+from azure.storage.blob import BlobClient, BlobProperties, ContainerClient
 from azure.storage.blob._models import BlobPrefix
 
 from keg_storage import backends
 from keg_storage.backends import base
 
 
-@mock.patch.object(backends.AzureStorage, '_create_container_client')
-class TestAzureStorage:
-    def create_storage(self, **kwargs):
-        return backends.AzureStorage(**{
-            'account': 'foo',
-            'key': base64.b64encode(b'a' * 64).decode(),
-            'bucket': 'test',
-            **kwargs,
-        })
+def create_storage(**kwargs):
+    return backends.AzureStorage(**{
+        'account': 'foo',
+        'key': base64.b64encode(b'a' * 64).decode(),
+        'bucket': 'test',
+        **kwargs,
+    })
 
-    def test_list(self, m_client):
-        m_walk = m_client.return_value.walk_blobs
+
+class TestAzureStorageBasics:
```

Comment:
I reorganized the tests under a few different classes, so as not to apply mocks to broadly.